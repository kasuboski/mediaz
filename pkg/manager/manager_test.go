package manager

import (
	"bytes"
	"context"
	"encoding/json"
	"io"
	"net/http"
	"strconv"
	"testing"
	"testing/fstest"

	"github.com/kasuboski/mediaz/pkg/library"
	"github.com/kasuboski/mediaz/pkg/prowlarr"
	prowlMock "github.com/kasuboski/mediaz/pkg/prowlarr/mocks"
	"github.com/kasuboski/mediaz/pkg/storage"
	"github.com/kasuboski/mediaz/pkg/storage/sqlite"
	"github.com/kasuboski/mediaz/pkg/tmdb/mocks"
	"github.com/oapi-codegen/nullable"
	"github.com/stretchr/testify/assert"
	"go.uber.org/mock/gomock"
)

func TestAddMovietoLibrary(t *testing.T) {
	ctrl := gomock.NewController(t)
	tmdbMock := mocks.NewMockClientInterface(ctrl)
	tmdbMock.EXPECT().MovieDetails(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(mediaDetailsResponse("test movie", 120), nil).Times(1)

	prowlarrMock := prowlMock.NewMockClientInterface(ctrl)
	indexers := []Indexer{{ID: 1, Name: "test", Priority: 1}, {ID: 3, Name: "test2", Priority: 10}}
	prowlarrMock.EXPECT().GetAPIV1Indexer(gomock.Any()).Return(indexersResponse(t, indexers), nil).Times(1)

	bigSeeders := nullable.NewNullNullable[int32]()
	bigSeeders.Set(23)

	smallSeeders := nullable.NewNullNullable[int32]()
	smallSeeders.Set(10)

	releases := []*prowlarr.ReleaseResource{{ID: intPtr(123), Title: nullable.NewNullableWithValue("test movie"), Size: sizeGBToBytes(23)}, {ID: intPtr(124), Title: nullable.NewNullableWithValue("test movie - very small"), Size: sizeGBToBytes(1)}}
	prowlarrMock.EXPECT().GetAPIV1Search(gomock.Any(), gomock.Any()).Return(searchIndexersResponse(t, releases), nil).Times(len(indexers))

	store, err := sqlite.New(":memory:")
	assert.Nil(t, err)

	schemas, err := storage.ReadSchemaFiles("../storage/sqlite/schema/schema.sql", "../storage/sqlite/schema/defaults.sql")
	assert.Nil(t, err)

	ctx := context.Background()
	err = store.Init(ctx, schemas...)
	assert.Nil(t, err)

	movieFS := fstest.MapFS{}
	tvFS := fstest.MapFS{}
	lib := library.New(movieFS, tvFS)
	pClient, err := prowlarr.New(":", "1234")
	pClient.ClientInterface = prowlarrMock
	assert.Nil(t, err)
	m := New(tmdbMock, pClient, lib, store)
	assert.NotNil(t, m)

	req := AddMovieRequest{
		TMDBID:           1234,
		QualityProfileID: 1,
	}
	release, err := m.AddMovieToLibrary(ctx, req)
	assert.Nil(t, err)

	assert.NotNil(t, release)
	assert.Equal(t, int32(123), *release.ID)
}

func searchIndexersResponse(t *testing.T, releases []*prowlarr.ReleaseResource) *http.Response {
	resp := &http.Response{
		StatusCode: http.StatusOK,
		Header:     make(map[string][]string),
	}

	out, err := json.Marshal(releases)
	assert.Nil(t, err)

	resp.Body = io.NopCloser(bytes.NewBuffer(out))

	return resp
}

// mediaDetailsResponse returns an http.Response that represents a MediaDetails with the given title and runtime
func mediaDetailsResponse(title string, runtime int) *http.Response {
	resp := &http.Response{
		StatusCode: http.StatusOK,
		Header:     make(map[string][]string),
	}

	resp.Body = io.NopCloser(bytes.NewBufferString(`{"title":"` + title + `","runtime":` + strconv.Itoa(runtime) + `}`))
	return resp

}

func indexersResponse(t *testing.T, indexers []Indexer) *http.Response {
	resp := &http.Response{
		StatusCode: http.StatusOK,
		Header:     make(map[string][]string),
	}

	prowlarrIndexers := make([]*prowlarr.IndexerResource, len(indexers))
	for i, indexer := range indexers {
		prowlarrIndexers[i] = &prowlarr.IndexerResource{
			ID:           &indexer.ID,
			Name:         nullable.NewNullableWithValue(indexer.Name),
			Priority:     &indexer.Priority,
			Capabilities: &prowlarr.IndexerCapabilityResource{},
		}
	}
	out, err := json.Marshal(prowlarrIndexers)
	assert.Nil(t, err)
	resp.Body = io.NopCloser(bytes.NewBuffer(out))
	return resp
}

func sizeGBToBytes(gb int) *int64 {
	b := int64(gb * 1024 * 1024 * 1024)
	return &b
}
