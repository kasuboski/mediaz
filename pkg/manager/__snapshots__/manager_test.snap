[TestMovieRejectRelease/prefix_match_only - 1]
[]string{"Brothers 2024 720p [broski]", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 2]
[]string{"10 Brothers Of Shaolin [720] HD (1977)", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 3]
[]string{"Step Brothers 2008 2160p UNRATED Bluray x265 DDP Atmos DTS KiNGDOM", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 4]
[]string{"The-Brothers-Karamazov-1969-(Dostoevsky-Mini-Series)-1080p-BRRip-x264-Classics-[..", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 5]
[]string{"Brothers 2024 1080p AMZN WEB DLip ExKinoRay", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 6]
[]string{"Brothers 2024 XviD WEB DLRip Enfanloup", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 7]
[]string{"Brothers 2024 D WEB DLRip 1 46Gb MegaPeer", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 8]
[]string{"Brothers 2024 1080p RHS [scarfilm org]", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 9]
[]string{"Superior 8 Ultra Brothers (T N) mp4", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 10]
[]string{"Brothers.2024.1080p.AMZN.WEBRip.1400MB.DD5.1.x264-GalaxyRG", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 11]
[]string{"The Blues Brothers (1980) 1080p BrRip x264 -YIFY", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 12]
[]string{"The.Menendez.Brothers.2024.1080p.WEBRip.1400MB.DD5.1.x264-Galaxy", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 13]
[]string{"Brothers.2024.720p.AMZN.WEBRip.800MB.x264-GalaxyRG", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 14]
[]string{"The.Menendez.Brothers.2024.720p.WEBRip.800MB.x264-GalaxyRG", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 15]
[]string{"The Miranda Brothers (2024) Hindi 1080p HDRip x264 AAC 5.1 ESubs [2GB] -QRips", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 16]
[]string{"Step Brothers 2008 UNRATED 1080p BluRay HEVC x265 5.1 BONE", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 17]
[]string{"The.Brothers.Karamazov.1969.(Dostoevsky-Mini.Series).1080p.BRRip.x264-Classics", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 18]
[]string{"Brothers (2009) 720p BrRip x264 - 700MB -YIFY", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 19]
[]string{"Brothers.2024.1080p.AMZN.WEBRip.DDP5.1.x265.10bit-GalaxyRG265", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 20]
[]string{"The Brothers Grimsby 2016 1080p BluRay x264 DTS-JYK", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 21]
[]string{"The.Menendez.Brothers.2024.1080p.WEB.h264-ETHEL[TGx]", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 22]
[]string{"Four Brothers (2005) 1080p BrRip x264 - 1.45GB -YIFY", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 23]
[]string{"The Brothers Grimm (2005) 720p BrRip x264 - 700MB -YIFY", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 24]
[]string{"Step Brothers.UNRATED.2008.1080p.BluRay.H264.ACC.5.1.BADASSMEDIA", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 25]
[]string{"The.Blues.Brothers.2000.1998.1080p.BluRay.H264.AAC", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 26]
[]string{"Brothers.2024.1080p.WEB.H264-AccomplishedYak", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 27]
[]string{"Rocco.and.His.Brothers.1960.SC.1080p.BluRay.x265.HEVC.AAC-SARTRE", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 28]
[]string{"Brothers 2024 720p AMZN WEB-DL DDP5 1 Atmos H 264-FLUX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 29]
[]string{"Four Brothers (2005) 720p BrRip x264 - 700MB-YIFY", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 30]
[]string{"The.Blues.Brothers.1980.EXTENDED.720p.BluRay.999MB.x265.10bit", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 31]
[]string{"Step Brothers 2008 Extended REPACK 1080p BluRay DD  7 1 X265-Ralphy", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 32]
[]string{"The Brothers Grimsby 2016 720p BluRay DTS x264-FuzerHD", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 33]
[]string{"The Brothers Bloom (2008) 1080p BrRip x264 -YIFY", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 34]
[]string{"Brothers.2024.1080p.10bit.WEBRip.6CH.x265.HEVC-PSA", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 35]
[]string{"Step Brothers (2008) 720p BrRip mkv - 550MB -YIFY", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 36]
[]string{"The.Sisters.Brothers.2018.1080p.WEB-DL.DD5.1.H264-CMRG[EtHD]", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 37]
[]string{"The.Brothers.Grimm.2005.1080p.AMZN.WEB-DL.DDP.5.1.H.264-PiRaTeS", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 38]
[]string{"The.Brothers.Grimm.2005.1080p.PMTP.WEB-DL.DDP.5.1.H.264-PiRaTeS", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 39]
[]string{"The.Everly.Brothers.Harmonies.from.Heaven.2016.1080p.WEBRip.x264-CBFM", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 40]
[]string{"10.Brothers.Of.Shaolin.1977.720p.HD", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 41]
[]string{"10.Brothers.Of.Shaolin.1977.720p.HD", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 42]
[]string{"The.Sisters.Brothers.2018.1080p.BluRay.x264-OFT", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 43]
[]string{"The.Brothers.Grimm.2005.1080p.BluRay.x264-OFT", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 44]
[]string{"Brothers.2009.BluRay.1080p.REMUX.AVC.DTS-HD.MA.5.1-LEGi0N", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 45]
[]string{"The.Brothers.Grimm.2005.1080p.BluRay.DD+.5.1.X265-Ralphy", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 46]
[]string{"The.Wonderful.World.of.the.Brothers.Grimm.1962.SMiLEBOX.720p.BluRay.x264-GUACAMOLE", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 47]
[]string{"The.Wonderful.World.of.the.Brothers.Grimm.1962.SMiLEBOX.1080p.BluRay.x264-GUACAMOLE", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 48]
[]string{"The.Wonderful.World.of.the.Brothers.Grimm.1962.LETTERBOXED.720p.BluRay.x264-GUACAMOLE", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 49]
[]string{"The.Wonderful.World.of.the.Brothers.Grimm.1962.LETTERBOXED.1080p.BluRay.x264-GUACAMOLE", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 50]
[]string{"Oh.Brothers.2003.1080p.VIKI.WEB-DL.AAC2.0.H.264-DUSKLiGHT", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 51]
[]string{"Half.Brothers.2020.720p.BluRay.DD5.1.x264-iFT", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 52]
[]string{"Brothers.2024.2160p.WEB.HDR10.DDP5.1.Atmos.AV1-R.and.H", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 53]
[]string{"Brothers.2024.1080p.AMZN.WEB-DL.DDP5.1.H.265", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 54]
[]string{"Brothers.Nest.2018.720p.WEB.H264-DiMEPiECE", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 55]
[]string{"Brothers.Nest.2018.1080p.WEB.H264-DiMEPiECE", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 56]
[]string{"Scooby-Doo.Meets.the.Boo.Brothers.1987.1080p.BluRay.DDP.2.0.10bit.H.265-iVy", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 57]
[]string{"Step.Brothers.2008.1080p.BluRayRip.EAC3.5.1.x265-Lootera", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 58]
[]string{"Brothers.by.Blood.2020.1080p.BluRay.x264-OFT", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 59]
[]string{"Brothers.2009.1080p.BluRay.x264-OFT", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 60]
[]string{"The.Menendez.Brothers.2024.1080p.WEBRip.DDP.5.1.10bit.H.265-iVy", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 61]
[]string{"Brothers.2009.1080p.BluRay.DDP.5.1.10bit.H.265-iVy", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 62]
[]string{"Brothers.2009.1080p.BluRay.DDP.5.1.10bit.H.265-iVy", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 63]
[]string{"Blues.Brothers.2000.1998.1080p.BluRay.REMUX.AVC.DTS-HD.MA.5.1-ATELiER", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 64]
[]string{"Brothers.2024.1080p.WEBRip.DDP.Atmos.5.1.10bit.H.265-iVy", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 65]
[]string{"Brothers.2024.1080p.WEBRip.DDP.Atmos.5.1.10bit.H.265-iVy", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 66]
[]string{"Brothers.2024.1080p.WEBRip.DDP.Atmos.5.1.10bit.H.265-iVy", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 67]
[]string{"The.Sisters.Brothers.2018.2160p.UHD.BluRay.DTS-HD.MA.5.1.DV.HDR10.10Bit.x265-ZAX", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 68]
[]string{"The.Miranda.Brothers.2024.1080p.WebRip.EAC3.5.1.x265-Lootera", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 69]
[]string{"The.Miranda.Brothers.2024.1080p.WebRip.EAC3.5.1.x265-Lootera", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 70]
[]string{"Step.Brothers.2008.Extended.REPACK.1080p.BluRay.DD.7.1.X265-Ralphy", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 71]
[]string{"The.Menendez.Brothers.2024.web.netflix.sdr.documentary.1080p.av1.5.1.eac3-Rosy", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 72]
[]string{"Brothers.2024.MULTi.HDR.2160p.WEB.H265-UKDHD", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 73]
[]string{"Brothers.2024.MULTi.1080p.WEB.H264-UKDHD", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 74]
[]string{"Brothers.2024.1080p.DS4K.WEBRip.DD.5.1.X265-Ralphy", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 75]
[]string{"Brothers.2024.1080p.DS4K.WEBRip.DD.5.1.X265-Ralphy", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 76]
[]string{"Brothers.2024.1080p.AMZN.WEBRip.DDP5.1.x265.10bit-LAMA", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 77]
[]string{"Brothers.2024.1080p.AMZN.WEBRip.DDP5.1.x265.10bit-GalaxyRG265", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 78]
[]string{"Brothers.2024.[2160p].[4K].[WEB].[5.1].[YTS.MX]", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 79]
[]string{"Brothers.2024.1080p.AMZN.WEB-DL.DDP5.1.H.264-APEX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 80]
[]string{"Brothers.2024.[1080p].[WEBRip].[x265].[10bit].[5.1].[YTS.MX]", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 81]
[]string{"Brothers.2024.NORDiC.2160p.SDR.WEB-DL.DDP5.1.H.265-NorTekst", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 82]
[]string{"Brothers.2024.1080p.AMZN.WEB-DL.DDP5.1.H.264-APEX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 83]
[]string{"Brothers.2024.[1080p].[WEBRip].[5.1].[YTS.MX]", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 84]
[]string{"Brothers.2024.[720p].[WEBRip].[YTS.MX]", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 85]
[]string{"Brothers.2024.1080p.AMZN.WEB-DL.DDP5.1.H264-FHC", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 86]
[]string{"Brothers.2024.iTA-ENG.WEBDL.1080p.x264-Dr4gon", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 87]
[]string{"Brothers.2024.720p.WEB-DL.DDP.5.1.iTA.ENG.H264-FHC", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 88]
[]string{"Brothers.2024.1080p.AMZN.WEBRip.DD5.1.x264-LAMA", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 89]
[]string{"Brothers.2024.1080p.AMZN.WEBRip.1400MB.DD5.1.x264-GalaxyRG", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 90]
[]string{"Brothers.2024.720p.AMZN.WEBRip.800MB.x264-GalaxyRG", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 91]
[]string{"Brothers.2024.720p.AMZN.WEBRip.x264-LAMA", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 92]
[]string{"Brothers.2024.720p.WEB-DL.H.264.Dual.YG", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 93]
[]string{"Brothers.2024.NORDiCSUBS.1080p.WEBRip.10Bit.x265-SWAXX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 94]
[]string{"Brothers.2024.1080p.WEB-DL.H.264.Dual.YG", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 95]
[]string{"Brothers.2024.iTA-ENG.WEBDL.1080p.x264-Dr4gon", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 96]
[]string{"Brothers.2024.1080p.AMZN.WEB-DL.DDP5.1.H264-FHC", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 97]
[]string{"Brothers.2024.720p.WEB-DL.DDP.5.1.iTA.ENG.H264-FHC", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 98]
[]string{"Brothers.2024.1080p.WEB.H264-AccomplishedYak", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 99]
[]string{"Brothers.2024.2160p.AMZN.WEB-DL.DDP5.1.Atmos.H.265-FLUX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 100]
[]string{"Brothers.2024.2160p.AMZN.WEB-DL.DDP5.1.Atmos.HDR.H.265-FLUX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 101]
[]string{"Brothers.2024.1080p.AMZN.WEB-DL.DDP5.1.Atmos.H.264-FLUX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 102]
[]string{"Brothers.2024.720p.AMZN.WEB-DL.DDP5.1.Atmos.H.264-FLUX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 103]
[]string{"Brothers.2024.1080p.AMZN.WEB-DL.DDPA5.1.H.264-FLUX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 104]
[]string{"Brothers.2024.720p.AMZN.WEB-DL.DDPA5.1.H.264-FLUX", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 105]
[]string{"Brothers.2024.MULTI.2160p.WEB-DL.SDR.H265-AOC", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 106]
[]string{"Brothers.2024.2160p.WEB.H265-ImperialDashingGriffinOfHealth", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 107]
[]string{"Brothers.2024.HDR.2160p.WEB.H265-ImperialDashingGriffinOfHealth", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 108]
[]string{"Brothers.2024.1080p.WEB.H264-AccomplishedYak", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 109]
[]string{"The.Menendez.Brothers.2024.1080p.WEBRip.x265-DH", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 110]
[]string{"The.Menendez.Brothers.2024.DOC.MULTi.1080p.WEB.x264-DEAL", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 111]
[]string{"Brothers.Justice.2010.1080p.BluRay.x264-OFT", "false"}
---

[TestMovieRejectRelease/prefix_match_only - 112]
[]string{"Songs.My.Brothers.Taught.Me.2015.Kino.1080p.BluRay.x264-OFT", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 113]
[]string{"The.Menendez.Brothers.2024.1080p.NF.WEB-DL.DDP5.1.H.264-xblz", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 114]
[]string{"The.Menendez.Brothers.2024.720p.NF.WEB-DL.DDP5.1.H.264-Kitsune", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 115]
[]string{"Blues.Brothers.2000.1998.1080p.BluRay.x264-nikt0", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 116]
[]string{"The.Menendez.Brothers.2024.1080p.WEBRip.DD5.1.x264-LAMA", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 117]
[]string{"The.Menendez.Brothers.2024.1080p.WEBRip.1400MB.DD5.1.x264-GalaxyRG", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 118]
[]string{"The.Menendez.Brothers.2024.720p.WEBRip.x264-LAMA", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 119]
[]string{"The.Menendez.Brothers.2024.720p.WEBRip.800MB.x264-GalaxyRG", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 120]
[]string{"The.Menendez.Brothers.2024.720p.NF.WEB-DL.AAC5.1.H.264-PrimeFix", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 121]
[]string{"The.Menendez.Brothers.2024.1080p.WEB.h264-ETHEL", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 122]
[]string{"The.Menendez.Brothers.2024.1080p.WEBRip.5.1-LAMA", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 123]
[]string{"The.Menendez.Brothers.2024.720p.WEBRip-LAMA", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 124]
[]string{"The.Menendez.Brothers.2024.1080p.NF.WEB-DL.DDP5.1.H.264-FLUX", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 125]
[]string{"Step.Brothers.2008.2160p.MA.WEB-DL.DDP5.1.H.265-HHWEB", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 126]
[]string{"Inside.Lehman.Brothers.2018.1080p.BluRay.x264-BRMP", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 127]
[]string{"Flaming.Brothers.1987.1080p.Bluray.Remux.AVC.TrueHD.7.1-MiCR0", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 128]
[]string{"Step.Brothers.2008.Unrated.UHD.BluRay.2160p.TrueHD.Atmos.7.1.HEVC.REMUX-FraMeSToR", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 129]
[]string{"The.Sklar.Brothers.What.Are.We.Talking.About.2014.1080p.WEBRip-LAMA", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 130]
[]string{"The.Sklar.Brothers.What.Are.We.Talking.About.2014.720p.WEBRip-LAMA", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 131]
[]string{"The.Sklar.Brothers.What.Are.We.Talking.About.2014.1080p.AMZN.WEB-DL.DDP2.0.H.264-TEPES", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 132]
[]string{"Rocco.and.His.Brothers.1960.Bluray.Remux.AVC.FLAC.2.0", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 133]
[]string{"Operation.Brothers.2019.1080p.NF.WEB-DL.DDP5.1.x264-MZABI", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 134]
[]string{"Jiang.Hu.Aka.Blood.Brothers.2004.DVDRip.x264-HANDJOB", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 135]
[]string{"The.Blues.Brothers.1980.Unrated.1080p.AMZN.WEB-DL.DDP.5.1.H.264-PiRaTeS", "true"}
---

[TestMovieRejectRelease/prefix_match_only - 136]
[]string{"Scooby-Doo.Meets.the.Boo.Brothers.1987.1080p.BluRay.DDP.2.0.H.265-iVy", "true"}
---
